packages/api/AlvTime.Business/InvoiceRate/InvoiceRateService.cs

```diff
@@ -0,0 +1,86 @@
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+using AlvTime.Business.Holidays;
+using AlvTime.Business.Options;
+using AlvTime.Business.Utils;
+using Microsoft.Extensions.Options;
+
+namespace AlvTime.Business.InvoiceRate;
+
+public class InvoiceRateService
+{
+    private readonly IInvoiceRateRepository _invoiceRateRepository;
+    private readonly IRedDaysService _redDaysService;
+    private readonly TimeEntryOptions _timeEntryOptions;
+
+    public InvoiceRateService(IInvoiceRateRepository invoiceRateRepository,
+        IRedDaysService redDaysService,
+        IOptionsMonitor<TimeEntryOptions> optionsMonitor
+    )
+    {
+        _invoiceRateRepository = invoiceRateRepository;
+        _redDaysService = redDaysService;
+        _timeEntryOptions = optionsMonitor.CurrentValue;
+    }
+
+    public async Task<decimal> GetEmployeeInvoiceRateForPeriod(int userId, DateTime fromDate, DateTime toDate)
+    {
+        var userTasks = await _invoiceRateRepository.GetInvoiceRateStatistics(userId, fromDate, toDate);
+        var availableHoursWithoutVacation = GetUserAvailableHours(fromDate, toDate.AddHours(23).AddMinutes(59));
+
+        var taskDictionary = userTasks.GroupBy(GetTaskType)
+                                      .ToDictionary(taskGroup => taskGroup.Key);
+
+        decimal billableHours = taskDictionary.ContainsKey(TaskType.BILLABLE) ? taskDictionary[TaskType.BILLABLE].Sum(task => task.Value) : 0;
+        decimal vacationHours = taskDictionary.ContainsKey(TaskType.VACATION) ? taskDictionary[TaskType.VACATION].Sum(task => task.Value) : 0;
+
+        var availableHours = availableHoursWithoutVacation - vacationHours;
+
+        if (availableHours <= 0)
+        {
+            return billableHours / 1;
+        }
+
+        return billableHours / availableHours;
+    }
+
+
+    private TaskType GetTaskType(InvoiceRateResponseDto entry)
+    {
+        if (entry.TaskId == _timeEntryOptions.PaidHolidayTask)
```

Comment:
Det er noen som har ført på UnpaidHolidayTask for lenge lenge siden. Veldig edge case, men kan gi litt feil faktureringsgrad for 2-3 personer i Alv hvis de sjekker 2 år tilbake i tid :D

Reply:
Eller customer er jo uansett Alv da, så det vil bli kategorisert som NON_BILLABLE

Reply:
Du har et poeng. Den ferien må med, eller så blir raten feil :thinking: